### PR TITLE
Implement Customer Import

### PR DESCRIPTION
**Pull request by x4d3**

### CODE REVIEW

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<div id="batch-upload">
+  <div class="row">
+    <div class="col-xs-12">
+      <h1 translate>mnoe_admin_panel.dashboard.customers.batch_upload.batch_upload</h1>
+      <a ui-sref="dashboard.customers" class="text-muted">
+        <i class="fa fa-chevron-left"></i><span translate>mnoe_admin_panel.dashboard.customers.batch_upload.back</span>
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: Add a `&nbsp;` between the `i` and `span`

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<div id="batch-upload">
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: Add a `top-buffer-1` class

File: src/app/views/customers/customers.html [hunk 3]
```diff
@@ -21,8 +21,9 @@
           {{'mnoe_admin_panel.dashboard.customers.main.new_customer' | translate}} &nbsp;<span class="caret"></span>
         </button>
         <ul class="dropdown-menu" uib-dropdown-menu role="menu" aria-labelledby="simple-btn-keyboard-nav">
-          <li role="menuitem" ng-if="::main.adminConfig.isRegistrationEnabled()"><a href="" ng-click="vm.inviteUserModal()" translate>mnoe_admin_panel.dashboard.customers.main.invite_customer</a></li>
+          <li role="menuitem" ng-if="vm.isRegistrationEnabled"><a href="" ng-click="vm.inviteUserModal()" translate>mnoe_admin_panel.dashboard.customers.main.invite_customer</a></li>
           <li role="menuitem"><a ui-sref="dashboard.customers.create-step-1" translate>mnoe_admin_panel.dashboard.customers.main.create_customer</a></li>
+          <li role="menuitem" ng-if="vm.isCustomerBatchImportEnabled"><a ui-sref="dashboard.customers.batch-upload" translate>mnoe_admin_panel.dashboard.customers.main.batch_upload</a></li>
```
Comment (alexnoox) [inline src/app/views/customers/customers.html]: Add `::vm...` in all the `ng-if` (single binding)

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+              <input type="file" accept=".csv" ngf-select ng-model="vm.file" name="file" ngf-max-size="20MB" required ngf-model-invalid="errorFile">
+              <i ng-show="vm.form.file.$error.maxSize"><span translate>mnoe_admin_panel.dashboard.customers.batch_upload.file_too_large</span> : {{errorFile.size / 1000000|number:1}}MB, max 20M</i>
+              <div class="top-buffer-1">
+                <button type="button" class="btn btn-secondary" ng-click="vm.downloadExample()" translate>
+                  mnoe_admin_panel.dashboard.customers.batch_upload.download_example
+                </button>
+                <button type="button" class="btn btn-primary" ng-disabled="!vm.form.$valid || vm.isUploading" ng-click="vm.uploadFile()" translate>
+                  mnoe_admin_panel.dashboard.customers.batch_upload.submit
+                </button>
+                <span ng-show="vm.isUploading"><i class="fa fa-spinner fa-pulse fa-fw"></i></span>
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: This spinner should be in the disabled submit button.

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            </fieldset>
+            <div class="progress top-buffer-1" ng-show="vm.file.progress >= 0">
+              <uib-progressbar value="vm.file.progress" type="{{vm.progressBarType}}">
+                <span ng-show="vm.errors" translate>mnoe_admin_panel.dashboard.customers.batch_upload.upload_failed</span>
+                <span ng-show="vm.file.result" translate>mnoe_admin_panel.dashboard.customers.batch_upload.upload_successful</span>
+                <span ng-show="!vm.file.result && !vm.errors">{{vm.file.progress}}%</span>
+              </uib-progressbar>
+            </div>
+          </form>
+          <div class="bs-row">
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: Do not use a `.bs-row` use `.row`

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                <mnoe-users-local-list list="vm.addedUsers" view="all"></mnoe-users-local-list>
+              </div>
+              <div ng-show="vm.updatedUsers.length > 0">
+                <h3 translate>mnoe_admin_panel.dashboard.customers.batch_upload.updated_users</h3>
+                <mnoe-users-local-list list="vm.updatedUsers" view="all"></mnoe-users-local-list>
+              </div>
+            </div>
+          </div>
+
+          <div ng-show="vm.errors">
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: Use a `col-sm-12`

File: src/app/components/mno-file-uploader/mno-file-uploader.coffee [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  ''',
+  controller: ($timeout, Upload, MnoErrorsHandler) ->
+    ctrl = this
+
+    ctrl.uploadFile = ->
+      ctrl.startCallback()
+      ctrl.progressBarType = 'primary'
+      ctrl.isUploading = true
+      file = ctrl.file
+      form = ctrl.form
```
Comment (alexnoox) [inline src/app/components/mno-file-uploader/mno-file-uploader.coffee]: Use directly `ctrl.file` and `ctrl.form` everywhere

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    <div class="col-xs-12">
+      <h1 translate>mnoe_admin_panel.dashboard.customers.batch_upload.batch_upload</h1>
+      <a ui-sref="dashboard.customers" class="text-muted">
+        <i class="fa fa-chevron-left"></i>&nbsp;<span translate>mnoe_admin_panel.dashboard.customers.batch_upload.back</span>
+      </a>
+      <!-- File upload section -->
+      <mno-widget icon="fa-upload" heading="{{'mnoe_admin_panel.dashboard.customers.batch_upload.file_upload' | translate}}">
+        <mno-widget-header></mno-widget-header>
+        <mno-widget-body>
+          <div class="row col-sm-12">
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: No `row` and `col` on the same element

Comment (alexnoox) [reply src/app/views/customers/batch-upload/batch-upload.html]:
```
<div class="row">
   <div class="col-sm-12">...</div>
</div>
```

File: src/app/views/customers/customers.html [hunk 9]
```diff
@@ -21,8 +21,9 @@
           {{'mnoe_admin_panel.dashboard.customers.main.new_customer' | translate}} &nbsp;<span class="caret"></span>
         </button>
         <ul class="dropdown-menu" uib-dropdown-menu role="menu" aria-labelledby="simple-btn-keyboard-nav">
-          <li role="menuitem" ng-if="::main.adminConfig.isRegistrationEnabled()"><a href="" ng-click="vm.inviteUserModal()" translate>mnoe_admin_panel.dashboard.customers.main.invite_customer</a></li>
+          <li role="menuitem" ng-if="::vm.isRegistrationEnabled"><a href="" ng-click="::vm.inviteUserModal()" translate>mnoe_admin_panel.dashboard.customers.main.invite_customer</a></li>
```
Comment (alexnoox) [inline src/app/views/customers/customers.html]: You don't need the `::` on click, it is not watched, it triggers an event.

**Comment by x4d3** [comment]
@alexnoox I've applied the requested changes

File: src/app/views/customers/batch-upload/batch-upload.html [hunk 10]
```diff
@@ -9,11 +9,13 @@ <h1 translate>mnoe_admin_panel.dashboard.customers.batch_upload.batch_upload</h1
...
-            </button>
+          <div class="row">
+            <div class="col-sm-12">
+              <p translate>mnoe_admin_panel.dashboard.customers.batch_upload.description</p>
+              <button type="button" class="btn btn-secondary" ng-click="vm.downloadExample()" translate>
+                mnoe_admin_panel.dashboard.customers.batch_upload.download_example
+              </button>
+            </div>
           </div>
           <div class="row col-sm-12 top-buffer-1">
```
Comment (alexnoox) [inline src/app/views/customers/batch-upload/batch-upload.html]: `row` and `col-sm-12` are on the same element

**Pull request merged**